src/js/player.js

```diff
@@ -1208,6 +1221,46 @@ class Player extends Component {
       this.hasStarted(false);
       this.trigger('loadstart');
     }
+
+    // autoplay happens after loadstart for the browser,
+    // so we mimic that behavior
+    this.manualAutoplay_(this.autoplay());
+  }
+
+  /**
+   * Handle autoplay string values, rather than the typical boolean
+   * values that should be handled by the tech. Note that this is not
+   * part of any specification. Valid values and what they do can be
+   * found on the autoplay getter at Player#autoplay()
+   */
+  manualAutoplay_(type) {
+    if (!this.tech_ || typeof type !== 'string') {
+      return;
+    }
+
+    if (type === 'muted') {
+      this.muted(true);
+    }
+
+    // default to an object so that we can check
+    // if "then" is a property on it more easily
+    const playPromise = this.play();
+
+    if (!playPromise || !playPromise.then || !playPromise.catch) {
+      return;
+    }
+
+    playPromise.then(() => {
+      this.trigger('autoplay-success');
+    }).catch((e) => {
+      // if we are muted or not set to any here there is nothing we can do
+      if (type !== 'any' || this.muted()) {
+        this.trigger('autoplay-failure');
+        return;
```

Comment:
If we've muted the player and the autoplay still fails, shouldn't we unmute again so a subsequent play triggered by a human has sound?

Reply:
This sounds reasonable to me.